server/text_generation_server/utils/flash_attn.py

```diff
@@ -2,24 +2,54 @@
 import torch
 
 from loguru import logger
-import math
 
-from text_generation_server.utils.import_utils import (
-    IS_CUDA_SYSTEM,
-    IS_ROCM_SYSTEM,
-    IS_XPU_SYSTEM,
-)
+from text_generation_server.utils.import_utils import SYSTEM
 
 if os.getenv("USE_FLASH_ATTENTION", "").lower() == "false":
     raise ImportError("`USE_FLASH_ATTENTION` is false.")
 HAS_FLASH_ATTN = True
 HAS_FLASH_ATTN_V2_CUDA = False
 HAS_FLASH_ATTN_V2_ROCM = False
 
-if IS_XPU_SYSTEM:
+if SYSTEM == "xpu":
     import intel_extension_for_pytorch as ipex
 
-if IS_CUDA_SYSTEM or IS_ROCM_SYSTEM:
+    def attention(
+        q,
```

Comment:
I suggested to do this kind of thing at some point but @OlivierDehaene did disagree with this design: https://github.com/huggingface/text-generation-inference/pull/1243#discussion_r1386307382

```
if IS_X_BACKEND:
    def foo():
        ...
else:
    def foo():
        ...